react/src/components/UserResourcePolicyList.tsx

```diff
@@ -1,15 +1,320 @@
+import React, { useState, useTransition } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLazyLoadQuery, useMutation } from 'react-relay';
 
-interface UserResourcePolicyListProps {
-  // Define your prop types here
-}
+type UserResourcePolicies = NonNullable<
+  UserResourcePolicyListQuery$data['user_resource_policies']
+>[number];
 
-const UserResourcePolicyList: React.FC<UserResourcePolicyListProps> = (
-  props,
-) => {
-  // Component logic goes here
+interface UserResourcePolicyListProps {}
 
-  return <div>User</div>;
+const UserResourcePolicyList: React.FC<UserResourcePolicyListProps> = () => {
+  const { token } = theme.useToken();
+  const { t } = useTranslation();
+  const { message } = App.useApp();
+
+  const [isRefetchPending, startRefetchTransition] = useTransition();
+  const [userResourcePolicyFetchKey, updateUserResourcePolicyFetchKey] =
+    useUpdatableState('initial-fetch');
+  const [isCreatingPolicySetting, setIsCreatingPolicySetting] = useState(false);
+  const [isOpenColumnsSetting, setIsOpenColumnsSetting] = useState(false);
+  const [inFlightResourcePolicyName, setInFlightResourcePolicyName] =
+    useState<string>();
+  const [editingUserResourcePolicy, setEditingUserResourcePolicy] =
+    useState<UserResourcePolicySettingModalFragment$key | null>();
+
+  const baiClient = useSuspendedBackendaiClient();
+  const supportMaxVfolderCount = baiClient?.supports(
+    'max-vfolder-count-in-user-and-project-resource-policy',
+  );
+  const supportMaxQuotaScopeSize = baiClient?.supports('max-quota-scope-size');
+  const supportMaxSessionCountPerModelSession = baiClient?.supports(
+    'max-session-count-per-model-session',
+  );
+  const supportMaxCustomizedImageCount = baiClient?.supports(
+    'max-customized-image-count',
+  );
+
+  const { user_resource_policies } =
+    useLazyLoadQuery<UserResourcePolicyListQuery>(
+      graphql`
+        query UserResourcePolicyListQuery {
+          user_resource_policies {
+            id
+            name
+            created_at
+            max_vfolder_count @since(version: "23.09.6")
+            max_quota_scope_size @since(version: "24.03.1")
+            max_session_count_per_model_session @since(version: "23.09.10")
+            max_customized_image_count @since(version: "24.03.0")
+            ...UserResourcePolicySettingModalFragment
+          }
+        }
+      `,
+      {},
+      {
+        fetchPolicy:
+          userResourcePolicyFetchKey === 'initial-fetch'
+            ? 'store-and-network'
+            : 'network-only',
+        fetchKey: userResourcePolicyFetchKey,
+      },
+    );
+
+  const [commitDelete, isInflightDelete] =
+    useMutation<UserResourcePolicyListMutation>(graphql`
+      mutation UserResourcePolicyListMutation($name: String!) {
+        delete_user_resource_policy(name: $name) {
+          ok
+          msg
+        }
+      }
+    `);
+
+  const columns: ColumnsType<UserResourcePolicies> = _.filter([
```

Comment:
With the type definition below, you can remove all `UserResourcePolicies` on every `sorter` and `render`. Be careful, The type (T) of `_.filter<T>` is not an array type, but an item type.
```suggestion
  const columns = _.filter<ColumnType<UserResourcePolicies>>([
```